pkg/controller/build/build_controller.go

```diff
@@ -72,5 +78,116 @@ type ReconcileBuild struct {
 func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling Build")
+
+	b := &build.Build{}
+	err := r.client.Get(context.Background(), request.NamespacedName, b)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Validate if the spec.output.secretref exist in the namespace
+	if b.Spec.Output.SecretRef != nil {
+		if b.Spec.Output.SecretRef.Name != "" {
```

Comment:
Any reason this is not a single liner 
```
b.Spec.Output.SecretRef != nil && b.Spec.Output.SecretRef.Name 
```
?

Reply:
no reason, I just modify this.